### ISSUE TITLE
JStock may be violating the GPL with a closed-source component

### ISSUE THREAD

**Issue by keilw** [yccheok/jstock#51]
Hi,

A package org.yccheok.jstock.engine is closed source component, while the rest of JStock claims to use the GPL. I am not a lawyer though involved in many license discussions especially via JCP EC.
GPL is very restrictive as to where closed source components may be used together with a GPL project, see http://www.quora.com/Is-it-OK-to-use-closed-source-libraries-in-an-open-source-program.

At first sight, I would say, the " org.yccheok.jstock.engine" package not qualifying to be a "system library" it likely violates using GPL for the entire project. Please check and consider your license or making " org.yccheok.jstock.engine" available Open Source, too.

**Comment by yccheok** [comment]
Thanks for input. org.yccheok.jstock.engine is still open source component (Hint: You can look at its history) not long ago. To avoid involving myself into some legal issues, I decide make it as a closed source component.

I'm also not clear whether GPL code can include closed source code. I shall do some research over it...

I may open source "org.yccheok.jstock.engine" again. But, not in the near term.

**Comment by keilw** [comment]
Please do. I know, at least a case of Austrian insurance institution who violated GPL by not publishing something extending a GPL piece and FSF (Free Software Foundation) took them to court. It may not happen wiith every project, but better not take chances.

**Comment by abelgo** [comment]
Ok so how can I build the project If I can't get the source for org.yccheok.jstock.engine ??

**Comment by yccheok** [comment]
@abelgo The code before closed source can be browsed through here : https://github.com/yccheok/jstock/tree/c8b1e8acced496f60f5aade644b0d4bb845dff54/src/org/yccheok/jstock/engine

**Comment by guitarzero** [comment]
Hello
Where is it possible to download this closed-source component ? 
Thanks

**Comment by yccheok** [comment]
We host this closed source in gitlab. You may request to join our closed source project in gitlab.

**Comment by javabrett** [comment]
It's not clear to me how to join the closed-source project in gitlab - I wandered over there, but if it's closed-source ... it's not showing in public projects of course.

The pre-delete version of the engine-code is already too old to build `master` with e.g. missing `org.yccheok.jstock.engine.DispType` - is there a binary dependency that can be used ... maybe just mix-in jstock.jar from the current binary dist?

**Comment by rkfg** [comment]
Yes, this program is in violation of GPL terms. The GPL license is viral so everything that includes GPL code and is being distributed in binary form should be open to study and modifications by means of providing the source code. That said, the repository doesn't include a license file but the source files have the license in their headers. Now it seems that the essential part of the program, the core, is unavailable and violates the user's freedom number 1:

> The freedom to study how the program works, and change it so it does your computing as you wish. **Access to the source code is a precondition for this.**

Quoted from [gnu.org](https://www.gnu.org/philosophy/free-sw.html), accentuation is mine. Overall, you don't have to be a lawyer to check if you software is in agreement with GPL, just read the aforementioned article and answer to yourself whether your product complies with the four freedoms or not from the user's standpoint. The legal license text only translates these ideas to the legal language but of course, if unsure consult a lawyer.

>To avoid involving myself into some legal issues, I decide make it as a closed source component.

This is worrysome. If you've already violated someone's rights it's no use hiding the source. Better to resolve the issue (rewrite questionable parts and such) and make it free again. I really like the program and it's quite sad to find its source in this unusable state. Basically, it's the only one portfolio tracking & techanalysis alive FOSS program I could find. I hope its future will be secured.

**Comment by masteryod** [comment]
This looks like a great app but this closed source binary is worrisome. One of the many beneficial aspects of open source is being able to look what's inside. I'm not going to trust random app. I don't want my metadata collected/send/sold...
Please, please work this out, keep this app open source and you'll have another happy user :)

**Comment by throwaway12341234** [comment]
This looks like a great app but this closed source binary is worrisome. One of the many beneficial aspects of open source is being able to look what's inside. I'm not going to trust random app. I don't want my metadata collected/send/sold... Please, please work this out, keep this app open source and you'll have another happy user :)

**Comment by deragon** [comment]
This is disappointing.  I was looking forward to contribute to the project, but if it contains a closed source component, it turns me off.  Why does that component must be closed?  Is this some form of engine you use in other projects with which you gain a living with?  If such is the case, this is respectable and I understand you, but for jStock, it is not very helpful.  Can this closed source component be replaced by another open source engine?

**Comment by rkfg** [comment]
> Can this closed source component be replaced by another open source engine?

No, because the engine is basically the program itself, the computational core. Replacing it would mean rewriting the project almost from scratch.

**Comment by yccheok** [comment]
Due to complicated legal issue, we decides not to release certain code block.

We should release some binary library, to make the source code build-able.

However, we are pretty much busy with other higher priority thing. Hence, we don't have an estimated timeline on when the binary library will be available.
